src/ui/components/componentmanager.js

```diff
@@ -178,16 +185,18 @@ export default class ComponentManager {
   remove (component) {
     this._core.storage.removeListener(this._componentToModuleIdListener.get(component));
 
-    const index = this._activeComponents.findIndex(c => c.name === component.name);
-    this._activeComponents.splice(index, 1);
+    const index = this._activeComponents.findIndex(c => c.uuid === component.uuid);
+    if (index !== -1) {
+      this._activeComponents.splice(index, 1);
+    }
   }
 
   /**
    * Remove the component with the given name
    * @param {string} name The name of the compnent to remove
    */
   removeByName (name) {
-    const component = this._activeComponents.find(c => c.name === name);
+    const component = Object.values(this._activeComponents).find(c => c.name === name);
     component.remove();
```

Comment:
it's ridiculous but component.remove() ends up calling ComponentManager.remove(this, so we don't need to update this line